test/inspector/test-inspector-port-zero-cluster.js

```diff
@@ -4,31 +4,43 @@ const common = require('../common');
 
 common.skipIfInspectorDisabled();
 
+// Assert that even when started with `--inspect=0` workers are assigned
+// consecutive (i.e. deterministically predictable) debug ports
+
 const assert = require('assert');
 const cluster = require('cluster');
 
-if (cluster.isMaster) {
-  const ports = [];
-  for (const worker of [cluster.fork(),
-                        cluster.fork(),
-                        cluster.fork()]) {
-    worker.on('message', common.mustCall((message) => {
-      ports.push(message.debugPort);
-      worker.kill();
+function serialFork() {
+  return new Promise((res) => {
+    const worker = cluster.fork();
+    worker.on('exit', common.mustCall((code, signal) => {
+      // code 0 is normal
+      // code 12 can happen is inspector could not bind because of a port clash
+      if (code !== 0 && code !== 12)
+        assert.fail(`code: ${code}, signal: ${signal}`);
+      const port = worker.process.spawnargs
+        .map((a) => (/=(?:.*:)?(\d{2,5})$/.exec(a) || [])[1])
+        .filter((p) => p)
+        .pop();
+      res(Number(port));
     }));
-    worker.send('debugPort');
-  }
-  process.on('exit', () => {
-    ports.sort();
-    assert.strictEqual(ports.length, 3);
-    assert(ports.every((port) => port > 0));
-    assert(ports.every((port) => port < 65536));
-    assert.strictEqual(ports[0] + 1, ports[1]);  // Ports should be consecutive.
-    assert.strictEqual(ports[1] + 1, ports[2]);
   });
+}
+
+if (cluster.isMaster) {
+  Promise.all([serialFork(), serialFork(), serialFork()])
+    .then(common.mustCall((ports) => {
+      ports.push(process.debugPort);
+      ports.sort();
+      // 4 = [master, worker1, worker2, worker3].length()
+      assert.strictEqual(ports.length, 4);
+      assert(ports.every((port) => port > 0));
+      assert(ports.every((port) => port < 65536));
+      // Ports should be consecutive.
+      assert.strictEqual(ports[0] + 1, ports[1]);
+      assert.strictEqual(ports[1] + 1, ports[2]);
+      assert.strictEqual(ports[2] + 1, ports[3]);
+    })).catch((err) => console.error(err));
```

Comment:
Will this `catch()` cause the test to exit without an error code if there is a problem? In other words, should we perhaps throw or `asesrt.fail()` here rather than (or in addition to) logging to the console?